core/templates/dev/head/services/context.service.ts

```diff
@@ -225,6 +225,19 @@ export class ContextService {
       this.getPageContext() ===
         ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR);
   }
+
+  canAddOrEditComponents(): boolean {
+    return (
```

Comment:
Here is a cleaner way to do this: define a temporary local variable in this function to store this.getPageContext(), and create a list of the allowed page contexts. Then check whether the list contains the variable.

Reply:
done